Author: Mathias Krause <minipli@googlemail.com>
Description: VSOCK: Fix missing msg_namelen update in vsock_stream_recvmsg()
 The code misses to update the msg_namelen member to 0 and therefore
 makes net/socket.c leak the local, uninitialized sockaddr_storage
 variable to userland -- 128 bytes of kernel stack memory.

--- a/modules/linux/vsock/linux/af_vsock.c
+++ b/modules/linux/vsock/linux/af_vsock.c
@@ -5455,6 +5455,8 @@ VSockVmciStreamRecvmsg(struct kiocb *kio
    vsk = vsock_sk(sk);
    err = 0;
 
+   msg->msg_namelen = 0;
+
    lock_sock(sk);
 
    if (sk->sk_state != SS_CONNECTED) {
